compiler/semmagic.nim

```diff
@@ -13,6 +13,8 @@
 proc semAddr(c: PContext; n: PNode; isUnsafeAddr=false): PNode =
   result = newNodeI(nkAddr, n.info)
   let x = semExprWithType(c, n)
+  if x.kind == nkSym:
```

Comment:
On the other hand, maybe I (and Nim's js backend ;) ) don't fully understand what you mean by sfAddrTaken flag.